examples/hybrid_plugin_system.ipynb

```diff
@@ -0,0 +1,453 @@
+   "source": [
+    "can_resolve = py_manager.hasCapability(py_manager.Capability.kResolution)\n",
+    "\n",
+    "helpers.display_result(f\"Can resolve? {can_resolve}\")"
+   ],
+   "id": "27535337eda9f49e",
+   "outputs": [
+    {
+     "data": {
+      "text/markdown": "> **Result:**\n> `Can resolve? False`"
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "execution_count": 4
+  },
+  {
+   "metadata": {},
+   "cell_type": "markdown",
+   "source": "So no, the Python plugin does not support the `resolve` method, at least on its own...",
+   "id": "8b67117d2f42a87e"
+  },
+  {
+   "metadata": {},
+   "cell_type": "markdown",
+   "source": [
+    "### The hybrid plugin system\n",
+    "\n",
+    "Given the two `ManagerImplementationFactoryInterface` instances (`cpp_factory` and `python_factory`), we can create a hybrid factory."
+   ],
+   "id": "5fd4a729153276f9"
+  },
+  {
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2024-09-20T07:28:17.201569Z",
+     "start_time": "2024-09-20T07:28:17.198604Z"
+    }
+   },
+   "cell_type": "code",
+   "source": [
+    "from openassetio.pluginSystem import HybridPluginSystemManagerImplementationFactory\n",
+    "\n",
+    "\n",
+    "hybrid_factory = HybridPluginSystemManagerImplementationFactory(\n",
+    "    [cpp_factory, python_factory], logger)\n",
+    "\n",
+    "manager = ManagerFactory.defaultManagerForInterface(\n",
+    "    \"resources/hybrid_plugin_system/openassetio_config.toml\",\n",
+    "    host_interface,\n",
+    "    hybrid_factory,\n",
+    "    logger)"
+   ],
+   "id": "87c3d7e701b095d9",
+   "outputs": [],
+   "execution_count": 5
+  },
+  {
+   "metadata": {},
+   "cell_type": "markdown",
+   "source": [
+    "Success! \n",
+    "\n",
+    "Notice how only a single config file (`openassetio_config.toml`) was provided. With hybrid plugins, the same configuration file is used for all the constituent plugins. In particular, any and all manager settings specified in the config file are passed to all plugins during initialisation.\n",
+    "\n",
+    "Is this combined hybrid plugin now capable of resolution?"
+   ],
+   "id": "4924fc6cb8e0620f"
+  },
+  {
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2024-09-20T07:28:17.268483Z",
+     "start_time": "2024-09-20T07:28:17.265302Z"
+    }
+   },
+   "cell_type": "code",
+   "source": [
+    "can_resolve = manager.hasCapability(py_manager.Capability.kResolution)\n",
+    "\n",
+    "helpers.display_result(f\"Can resolve? {can_resolve}\")"
+   ],
+   "id": "2869873c40ecc805",
+   "outputs": [
+    {
+     "data": {
+      "text/markdown": "> **Result:**\n> `Can resolve? True`"
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "execution_count": 6
+  },
+  {
+   "metadata": {},
+   "cell_type": "markdown",
+   "source": "Again, success! The resolution capability from the C++ plugin has been combined with the capabilities of the Python plugin.",
```

Comment:
[consider] Is it worth re-iterating here that now any C++ host that calls resolve won't involve Python - and consequently the GIL - and so would remain multithreaded?

Reply:
Great shout - added a couple of sentences in ffede968